### PR TITLE
[#156313150] Remove migrate-cf-certs job, remove unnecesary bosh-CA certificate, unit test cert rotation

### PR DESCRIPTION
**Pull request by keymon**
[#156313150 Clean up work after migrating cert generation to bosh](https://www.pivotaltracker.com/story/show/156313150)

from [#155854402 Restructure internal certs to match cf-deployment](https://www.pivotaltracker.com/story/show/155854402)

What?
----

After merging alphagov/paas-cf#1292 and deploy it to production,
we have migrated the old cf-certs.tgz file to the new vars-store
file. The task migrate-cf-certs added to support a zero-downtime migration[1]
is not needed anymore

We also stop using stop using bosh-CA if not needed, after we stopped using it as CA of the [API and apps endpoints](https://github.com/alphagov/paas-cf/pull/1202) and for the [cf internal certs](https://github.com/alphagov/paas-cf/pull/1292)

Finally, we add a unit test to verify that every CA certificate used is modified to append the old certificate during rotation.

[1] b1abe8806bde01baa8ed3b3e20ae4af3722839bf

How to review?
-----------

 - Code review
 - Push the pipeline, it should work as usual
 - ~~Be sure that we migrated the certs in  prod and all the envs~~ (this is done)

Who?
----

Anyone but @keymon

### CODE REVIEW

File: manifests/cf-manifest/spec/support/manifest_helpers.rb [hunk 1]
```diff
@@ -52,6 +52,38 @@ def fetch(key, default_value = nil)
...
+          recursive_inject(acum2, x2, new_path) { |acum3, x3, path3|
+            yield(acum3, x3, path3)
+          }
+        }
+      else
+        yield(acum, x, path)
+      end
+    end
+
+    def inject(acum)
```
Comment (bandesz) [inline manifests/cf-manifest/spec/support/manifest_helpers.rb]: Can you please explain what is this used for and how is it used?

File: manifests/cf-manifest/spec/support/manifest_helpers.rb [hunk 2]
```diff
@@ -68,20 +100,30 @@ def workdir
     Cache.instance.workdir ||= $workdir
   end
 
+  def manifest_without_vars_store
```
Comment (bandesz) [inline manifests/cf-manifest/spec/support/manifest_helpers.rb]: This doesn't seem to be used anywhere

Comment (keymon) [reply manifests/cf-manifest/spec/support/manifest_helpers.rb]: I added a test in `manifests/cf-manifest/spec/manifest/rotate_certs_spec.rb`, I forgot to push, sorry.

File: manifests/cf-manifest/spec/support/manifest_helpers.rb [hunk 3]
```diff
@@ -68,20 +101,30 @@ def workdir
...
+    Cache.instance.manifest_without_vars_store ||= \
```
Comment (bandesz) [inline manifests/cf-manifest/spec/support/manifest_helpers.rb]: Originally this used ```Cache.instance.manifest_with_defaults``` which caused the test to use the wrong manifest.
I added a separate variable.

**Pull request merged**